action.yml

```diff
@@ -142,15 +142,18 @@ runs:
         # in the cache action:
         # https://github.com/actions/cache/issues/1241
 
-        if [ ! -d "/home/runner/.solcx" ]; then
-          mkdir "/home/runner/.solcx"
-          echo "Solcx directory created."
-        fi
-        if [ ! -d "/home/runner/.vvm" ]; then
-          mkdir "/home/runner/.vvm"
-          echo "VVM directory created."
-        fi
-        if [ ! -d "${{ github.workspace }}/.build" ]; then
-          mkdir "${{ github.workspace }}/.build"
-          echo ".build directory created."
+        # NOTE: If /home/runner does not exist, mkdir fails.
+        if [ -d "$HOME" ]; then
```

Comment:
ok turns out the problem is github-action never caches when using a mac runner because the name of the directory here was wrong. I am thinking changing it to $HOME variable will fix. I will know soon.